Source/WebCore/editing/TextIterator.cpp

```diff
@@ -772,7 +776,17 @@ bool TextIterator::handleReplacedElement()
 
     m_hasEmitted = true;
 
-    if (m_behaviors.contains(TextIteratorBehavior::EmitsObjectReplacementCharacters)) {
+    auto shouldEmitObjectReplacementCharacter = [&] {
+        if (m_behaviors.contains(TextIteratorBehavior::EmitsObjectReplacementCharacters))
+            return true;
+
+        if (is<HTMLImageElement>(m_currentNode.get()) && m_behaviors.contains(TextIteratorBehavior::EmitsObjectReplacementCharactersForImagesOnly))
```

Comment:
I would flip the order of conditions, `m_behaviors.contains` is a simpler check.